Makefile

```diff
@@ -0,0 +1,30 @@
+generate-code:
```

Comment:
A few updates to the makefile can help ensure that commands like `make test` can be run without explicitly having to run `make generate-code` before-hand.

Additionally, the `PHONY` make targets should be explicitly defined so that creating a file with the same name as one of these targets does not cause make to ignore running that target. For example, running `touch test` and then `make test` with this current Makefile would cause the tests not to be run. Adding a `test` as a `PHONY` target fixes this.

The below `Makefile` explicitly defines the `PHONY` targets and renames `generate-code` to `gen` then uses [order-only prerequisites](https://www.gnu.org/savannah-checkouts/gnu/make/manual/html_node/Prerequisite-Types.html) to ensure code generation is run if the `gen` folder does not already exist.  

```
.PHONY=deps lint test add-license check-license circleci-local shellcheck salus

deps:
	go get ./...
	go get golang.org/x/lint/golint
	go get github.com/google/addlicense

gen:
	./codegen.sh 

lint:
	golint -set_exit_status ./asserter/... ./fetcher/...

test: | gen
	go test -v ./asserter ./fetcher

LICENCE_SCRIPT=addlicense -c "Coinbase, Inc." -l "apache" -v

add-license:
	${LICENCE_SCRIPT} .

check-license:
	${LICENCE_SCRIPT} -check .

circleci-local:
	circleci local execute

shellcheck:
	shellcheck codegen.sh

salus:
	docker run --rm -t -v "${PWD}":/home/repo coinbase/salus
```

Reply:
It is often the case that we want to run `make gen` when the `gen` folder already exists (when the `spec.json` we are building from changes).